source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -142,7 +151,131 @@ private static string GetCompensationRequisitionStatusText(bool? isDraft)
             }
         }
 
-        private void AddNoteIfStatusChanged(long compensationRequisitionId, long acquisitionFileId, bool? currentStatus, bool? newStatus)
+        private static DateOnly? CheckFinalizedDate(bool? currentStatusIsDraft, bool? newStatusIsDraft, DateOnly? currentValue)
```

Comment:
nit: this function name says "Check..." but this returns a date Should probably be "Get..." or "Calculate..."

Reply:
renamed